runtime/lua/vim/secure.lua

```diff
@@ -21,6 +21,50 @@ local function read_trust()
   return trust
 end
 
+--- Reads the given file/directory at {fullpath} and returns the
+--- contents and a hash of the contents. If {fullpath} is not a directory
+--- and {bufnr} is given, then contents are read from the {bufnr} buffer rather
+--- than the filesytesm.
+---
+---@param fullpath (string) Path to a file or directory to read.
+---@param bufnr (number?) The number of the buffer.
+---@return string|boolean? contents the contents of the file, or true if it's a directory
+---@return string? hash the hash of the contents, or "directory" if it's a directory
+local function read_and_hash(fullpath, bufnr)
+  local contents ---@type string|boolean?
+  local hash ---@type string
+  if vim.fn.isdirectory(fullpath) == 1 then
+    contents = true
+    hash = "directory"
```

Comment:
Replace this with an early return, then the rest of the function doesn't need to go in an `else` block.

```suggestion
    return true, "directory"
```
```